typify-impl/src/test_util.rs

```diff
@@ -34,12 +34,25 @@ fn validate_output_impl<T: JsonSchema + Schema>(ignore_variant_names: bool) {
     let name = type_name::<T>().rsplit_once("::").unwrap().1.to_string();
 
     let mut type_space = TypeSpace::default();
+
+    // Convert all references
     type_space
         .add_ref_types(schema.definitions.clone())
         .unwrap();
-    let (ty, _) = type_space
-        .convert_schema_object(Name::Required(name), &schema.schema)
-        .unwrap();
+
+    // If we have converted the type already, use that, otherwise convert schema object
+    let ty = if let Some(already_type_id) = type_space.ref_to_id.get(&name) {
+        type_space
+            .id_to_entry
+            .get(&already_type_id)
+            .unwrap()
+            .clone()
+    } else {
+        let (ty, _) = type_space
+            .convert_schema_object(Name::Required(name), &schema.schema)
+            .unwrap();
+        ty
```

Comment:
```suggestion
       type_space
            .convert_schema_object(Name::Required(name), &schema.schema)
            .unwrap().0
```

Not sure if this is better or worse.

Reply:
Hmm, in this case I like the original way.

Reply:
Well good! I wrote that too! ;-)